File: 1-programming/1-intro/main.cpp
// main.cpp
// C++ example code demonstrating basic programming concepts.

#include <iostream>

// Function to print Hello World
void helloWorld() {
    std::cout << "Hello, World!" << std::endl;
}

// Function to add two numbers
int addNumbers(int a, int b) {
    return a + b;
}

// Recursive factorial function
int factorial(int n) {
    if (n == 0) {
        return 1;
    } else {
        return n * factorial(n - 1);
    }
}

int main() {
    helloWorld();
    std::cout << "Sum of 3 and 5: " << addNumbers(3, 5) << std::endl;
    std::cout << "Factorial of 5: " << factorial(5) << std::endl;
    return 0;
}

File: 1-programming/1-intro/main.java
// main.java
// Java example code demonstrating basic programming concepts.

class HelloWorld {
    
    // Method to print Hello World
    public static void helloWorld() {
        System.out.println("Hello, World!");
    }

    // Method to add two numbers
    public static int addNumbers(int a, int b) {
        return a + b;
    }

    // Recursive method to calculate factorial
    public static int factorial(int n) {
        if (n == 0) {
            return 1;
        } else {
            return n * factorial(n - 1);
        }
    }

    public static void main(String[] args) {
        helloWorld();
        System.out.println("Sum of 3 and 5: " + addNumbers(3, 5));
        System.out.println("Factorial of 5: " + factorial(5));
    }
}
